code/controllers/subsystem/migration.dm

```diff
@@ -310,11 +310,12 @@ SUBSYSTEM_DEF(migrants)
 	log_character("[human_character.ckey] ([fakekey]) - [human_character.real_name] - [migrant_job.title]")
 
 	var/datum/migrant_role/advclass/adv_migrant = role_instance
-	if(adv_migrant.advclass_cat_rolls)
-		SSrole_class_handler.setup_class_handler(human_character, adv_migrant.advclass_cat_rolls)
-		human_character.hugboxify_for_class_selection()
-	else if(GLOB.adventurer_hugbox_duration)
-		addtimer(CALLBACK(human_character, TYPE_PROC_REF(/mob/living/carbon/human, adv_hugboxing_start)), 1)
+	if(istype(adv_migrant))
```

Comment:
if adv_migrant is null or not a migrant_role something has to have gone wrong, right?

Reply:
oh nevermind